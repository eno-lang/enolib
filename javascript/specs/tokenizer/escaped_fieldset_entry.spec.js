const { inspectTokenization } = require('./util.js');

const input = `
fieldset:
\`key\` =

\`\`ke\`y\`\` =

\`\`\`k\`\`ey\`\`\`    =

    \`\` \`key\` \`\`    =

\`key\` =
`.trim();

describe('Escaped fieldset entry tokenization', () => {
  it('performs as specified', () => {
    expect(inspectTokenization(input)).toMatchSnapshot();
  });
});
