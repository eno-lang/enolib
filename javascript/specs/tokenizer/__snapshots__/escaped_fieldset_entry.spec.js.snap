// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Escaped fieldset entry tokenization performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [
      Object {
        "entries": Array [
          Object {
            "continuations": Array [],
            "key": "key",
            "line": 1,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                16,
                17,
              ],
              "escapeBeginOperator": Array [
                10,
                11,
              ],
              "escapeEndOperator": Array [
                14,
                15,
              ],
              "key": Array [
                11,
                14,
              ],
              "line": Array [
                10,
                17,
              ],
            },
            "type": Symbol(Attribute),
            "value": null,
          },
          Object {
            "continuations": Array [],
            "key": "ke\`y",
            "line": 3,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                28,
                29,
              ],
              "escapeBeginOperator": Array [
                19,
                21,
              ],
              "escapeEndOperator": Array [
                25,
                27,
              ],
              "key": Array [
                21,
                25,
              ],
              "line": Array [
                19,
                29,
              ],
            },
            "type": Symbol(Attribute),
            "value": null,
          },
          Object {
            "continuations": Array [],
            "key": "k\`\`ey",
            "line": 5,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                46,
                47,
              ],
              "escapeBeginOperator": Array [
                31,
                34,
              ],
              "escapeEndOperator": Array [
                39,
                42,
              ],
              "key": Array [
                34,
                39,
              ],
              "line": Array [
                31,
                47,
              ],
            },
            "type": Symbol(Attribute),
            "value": null,
          },
          Object {
            "continuations": Array [],
            "key": "\`key\`",
            "line": 7,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                68,
                69,
              ],
              "escapeBeginOperator": Array [
                53,
                55,
              ],
              "escapeEndOperator": Array [
                62,
                64,
              ],
              "key": Array [
                56,
                61,
              ],
              "line": Array [
                49,
                69,
              ],
            },
            "type": Symbol(Attribute),
            "value": null,
          },
          Object {
            "continuations": Array [],
            "key": "key",
            "line": 9,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                77,
                78,
              ],
              "escapeBeginOperator": Array [
                71,
                72,
              ],
              "escapeEndOperator": Array [
                75,
                76,
              ],
              "key": Array [
                72,
                75,
              ],
              "line": Array [
                71,
                78,
              ],
            },
            "type": Symbol(Attribute),
            "value": null,
          },
        ],
        "key": "fieldset",
        "line": 0,
        "parent": [Circular],
        "ranges": Object {
          "fieldOperator": Array [
            8,
            9,
          ],
          "key": Array [
            0,
            8,
          ],
          "line": Array [
            0,
            9,
          ],
        },
        "type": Symbol(Fieldset),
      },
    ],
    "type": Symbol(Document),
  },
  "_lineCount": 10,
  "_meta": Array [],
}
`;
